stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -5435,254 +7391,825 @@ fn missed_slots_no_unlock() {
-    // now check that bob has an unlock height of `height_target`
-    let bob_bal = get_stx_account_at(
-        &mut peer,
-        &latest_block,
-        &bob_address.to_account_principal(),
+// Helper struct to hold information about stackers and signers
+#[derive(Debug, Clone)]
+struct StackerSignerInfo {
+    private_key: StacksPrivateKey,
+    public_key: StacksPublicKey,
+    principal: PrincipalData,
+    address: StacksAddress,
+    pox_address: PoxAddress,
+    nonce: u64,
+}
+
+impl StackerSignerInfo {
+    fn new() -> Self {
+        let private_key = StacksPrivateKey::new();
+        let public_key = StacksPublicKey::from_private(&private_key);
+        let address = key_to_stacks_addr(&private_key);
+        let pox_address =
+            PoxAddress::from_legacy(AddressHashMode::SerializeP2PKH, address.bytes.clone());
+        let principal = PrincipalData::from(address.clone());
+        let nonce = 0;
+        Self {
+            private_key,
+            public_key,
+            address,
+            principal,
+            pox_address,
+            nonce,
+        }
+    }
+}
+
+/// Helper function to advance to a specific block height with the passed txs as the first in the block
+/// Returns a tuple of the tip and the observed block that should contain the provided txs
+fn advance_to_block_height(
+    peer: &mut TestPeer,
+    observer: &TestEventObserver,
+    txs: &[StacksTransaction],
+    peer_nonce: &mut usize,
+    target_height: u64,
+) -> (StacksBlockId, TestEventObserverBlock) {
+    let mut tx_block = None;
+    let mut latest_block = None;
+    let mut passed_txs = txs;
+    while peer.get_burn_block_height() < target_height {
+        latest_block = Some(peer.tenure_with_txs(&passed_txs, peer_nonce));
+        passed_txs = &[];
+        if tx_block.is_none() {
+            tx_block = Some(observer.get_blocks().last().unwrap().clone());
+        }
+    }
+    let latest_block = latest_block.expect("Failed to get tip");
+    let tx_block = tx_block.expect("Failed to get tx block");
+    (latest_block, tx_block)
+}
+// In this scenario, two service signers (Alice, Bob), one stacker-signer (Carl), two stacking pool operators (Dave, Eve), & six pool stackers (Frank, Grace, Heidi, Ivan, Judy, Mallory).
+
+// First Nakamoto Reward Cycle
+// First Nakamoto Tenure
+
+// 1. Franks stacks for 1 reward cycle, Grace stacks for 2 reward cycles & so on…Mallory stacks for 6 reward cycles: (so 6 wallets stacking n, n+1, n+2… cycles)
+// 2. Dave asks Alice for 3 signatures
+// 3. Eve asks Bob for 3 set-authorizations
+// 4. Ivan - Mallory ask Bob to set-approval-authorization
+// 5. Carl stx-stacks & self-signs for 3 reward cycle
+// 6. In Carl's second reward cycle, he calls stx-extend for 3 more reward cycles
+// 7. In Carl's third reward cycle, he calls stx-increase and should fail as he is straddling 2 keys
+#[test]
+fn test_scenario_five() {
```

Comment:
Mind adding more comemnts to scenario five test like you did with the other tests? makes it much easier to follow :D

Reply:
Added more comments for clarity!